### PR TITLE
RUBY-1056: Handle custom id generators

### PR DESCRIPTION
**Pull request by durran**
Can pass an option to client `:id_generator` that's value is an object that responds to `#generate`.

### CODE REVIEW

File: lib/mongo/collection.rb [hunk 1]
```diff
@@ -341,7 +341,7 @@ def insert_one(document, options = {})
           :coll_name => name,
           :write_concern => write_concern,
           :bypass_document_validation => !!options[:bypass_document_validation],
-          :options => options
+          :id_generator => client.options[:id_generator]
```
Comment (estolfo) [inline lib/mongo/collection.rb]: we are sure we don't need to pass :options to Operation::Write::Insert#initialize ?

File: spec/mongo/collection_spec.rb [hunk 2]
```diff
@@ -825,6 +856,37 @@
...
+
+      after do
+        custom_client.close
+      end
+
+      it 'inserts with the custom id' do
+        expect(custom_collection.find.first[:_id]).to eq(1)
+      end
+    end
+
```
Comment (estolfo) [inline spec/mongo/collection_spec.rb]: did you mean to repeat the spec?

Comment (durran) [inline spec/mongo/collection_spec.rb]: Yeah - one is insert many and one is insert one.

Comment (estolfo) [inline spec/mongo/collection_spec.rb]: ok, sorry, I missed that.

**Pull request merged**